src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -336,6 +343,12 @@ const bool Terminal::_isWordDelimiter(std::wstring_view cellChar) const
 // - the corresponding location on the buffer
 const COORD Terminal::_ConvertToBufferCell(const COORD viewportPos) const
 {
+    // don't change the value if at/outside the boundary
+    if (viewportPos.X <= 0 || viewportPos.X >= _buffer->GetSize().RightInclusive())

```

Comment:
Should we `THROW_HR(E_INVALIDARG)` in this case? Or have some other result to indicate it was outside the buffer?